src/session/session.c

```diff
@@ -399,7 +399,15 @@ exit_pam_init_problem (int result_code)
   else
     message = pam_strerror (NULL, result_code);
 
-  exit_init_problem (problem,  message);
+  /* we send the message through the Cockpit protocol, make sure we don't break JSON encoding */
+  char *message_sanitized = strdupx (message);
+  char *c;
+  while ((c = strchr (message, '"')) != NULL)
```

Comment:
You make a mutable copy of the string into `message_sanitized` and then you modify the original `message` before passing the unmodified copy to `exit_init_problem()` :)  `char*` unsafety issues strike again via the (terrible) signature of `strchr()` which takes a `const char *` and returns a `char *`.  I seriously start to hate C.

Shame we don't have [`g_strcanon`](https://docs.gtk.org/glib/func.strcanon.html) here.

The two passes over the string bother me and I'd have written it more like
```c
   for (int i = 0; message[i]; i++)
       if (message[i] != '\\') ... else if ...
```

but I bet yours is actually faster in practice.